contributors/design-proposals/scheduling/scheduling-policy.md

```diff
@@ -0,0 +1,516 @@
+
+### required
+Elements here are required, pods won't schedule if they aren't present. Also note that if something is required it is also allowed.
+
+### allowed
+Elements here are allowed, the policy allows the presence of these elements. From Pod's perspective, a pod can use one or N of the allowed items.
+
+### deny
+If pods specify one of these, the pod won't schedule to a node, we won't dive into deny as it is the exact opposite of required.
+
+### Scheduler name
+
+If `schedulerNames` is absent from `allowed`, `default` or `required`, no scheduler is allowed by this specific policy.
+
+#### required
+
+Require that pods use either the green-scheduler (which is used by specifying `spec.schedulerName` in pod definition) or the `my-scheduler` scheduler (by specifying `spec.schedulerName: "my-scheduler"`):
+
+```yaml
+apiVersion: policy/v1alpha1
+kind: SchedulingPolicy
+metadata:
+  name: mySchedulingPolicy
+spec:
+  bindingMode: all
+  namespaces:
+    - default
+  rules:
+    schedulerNames:
+      - match: ["green-scheduler","my-scheduler"]
+        action: required
+```
+
+An empty list of schedulerNames has no effect, as pod use the `defaultScheduler` if no scheduler is specified:
+
+```yaml
+apiVersion: policy/v1alpha1
+kind: SchedulingPolicy
+metadata:
+  name: mySchedulingPolicy
+spec:
+  bindingMode: all
+  namespaces:
+    - default
+  rules:
+    schedulerNames:
+      - match: []
+        action: required
+```
+
+
+#### allowed
+Allow pods to use either the green-scheduler (which is used by specifying `spec.schedulerName` in pod definition) or the `my-scheduler` scheduler (by specifying `spec.schedulerName: "my-scheduler"`) in the namespace `default`:
+
+```yaml
+apiVersion: policy/v1alpha1
+kind: SchedulingPolicy
+metadata:
+  name: mySchedulingPolicy
+spec:
+  bindingMode: all
+  namespaces:
+    - default
+  rules:
+    schedulerNames:
+      - match: ["green-scheduler","my-scheduler"]
+        action: allowed
+```
+
+An empty list of schedulerNames will allow usage of all schedulers:
+
+```yaml
+apiVersion: policy/v1alpha1
+kind: SchedulingPolicy
+metadata:
+  name: mySchedulingPolicy
+spec:
+  bindingMode: all
+  namespaces:
+    - default
+  rules:
+    schedulerNames:
+      - match: []
+        action: allowed
+```
+
+
+### Tolerations
+
+Toleration usage can be regulated using fine-grain rules with `tolerations` field. If specifying multiple `tolerations`, pod will be scheduled if one of the tolerations is satisfied.
+
+#### required
+
+This allows to require toleration in the following forms of:
+
+- tolerations that tolerates taints with key named `projectA-dedicated` with all effects.
+
+##### Fine-grain allowed tolerations
+
+```yaml
+apiVersion: policy/v1alpha1
+kind: SchedulingPolicy
+metadata:
+  name: mySchedulingPolicy
+spec:
+  bindingMode: all
+  namespaces:
+    - projectA
+  rules:
+    tolerations:
+      - match:
+        - keys: ["projectA-dedicated"]
+          operators: ["Exists"]
+          effects: []
```

Comment:
Using of array for `keys`, `operators` and `effects` are confusing. How to interpret the following examples:

```yaml
match:
- keys: ["projectA-dedicated", "projectB-dedicated"]
  operators: ["Exists"]
  effects: []
```

vs

```yaml
match:
- keys: ["projectA-dedicated", "projectB-dedicated"]
  operators: ["Exists", "NotExists"]
  effects: []
```

Is the mapping from `keys`, `operators` to `effects` element-by-element, or something else?

What about making it simpler, use one `operator` for all `keys`?

Reply:
I agree we should keep it in line with [label selectors](https://godoc.org/k8s.io/apimachinery/pkg/apis/meta/v1#LabelSelectorRequirement), unless there is precedent for something different.